### ISSUE TITLE
About loss function

### ISSUE THREAD

**Issue by liujiyuan13** [lukasruff/Deep-SVDD-PyTorch#8]
In paper, "Deep One-Class Classification",  the loss function goes as 
![image](https://user-images.githubusercontent.com/36629703/55679174-583e5a00-5939-11e9-9fc5-82d9a92a6b21.png)
However, the loss is calculated as 
![image](https://user-images.githubusercontent.com/36629703/55679181-70ae7480-5939-11e9-9e2c-ce17265a250b.png)
They are different in these two places. The loss in this code lacks of the weights of neural network. Is it important or considered in another place?

**Comment by rsaite** [comment]
The last term in the loss function is commonly referred to as weight decay and it is considered in another place. Its strength (lambda) is set through the parameters `--weight_decay` (and for pretraining `--ae_weight_decay`) which are passed to the optimizer.

**Comment by liujiyuan13** [comment]
> The last term in the loss function is commonly referred to as weight decay and it is considered in another place. Its strength (lambda) is set through the parameters `--weight_decay` (and for pretraining `--ae_weight_decay`) which are passed to the optimizer.

Thanks very much for your reply!
I've found the corresponding code for the weights of neural network.

**Comment by lukasruff** [comment]
As @rsaite pointed out, PyTorch makes it simple to add weight decay regularization on the network weights via the `optimizer`. Just to add the specific lines for reference:

**Deep SVDD trainer**
https://github.com/lukasruff/Deep-SVDD-PyTorch/blob/5d7195dfff37efdaccd337257484d9d3db464730/src/optim/deepSVDD_trainer.py#L49

**Autoencoder trainer for pretraining**
https://github.com/lukasruff/Deep-SVDD-PyTorch/blob/5d7195dfff37efdaccd337257484d9d3db464730/src/optim/ae_trainer.py#L30
